### PR TITLE
hand over ExtensionContext to GherkinListener

### PR DESCRIPTION
**Pull request by ArneAlbrechtW**

### CODE REVIEW

File: src/main/java/org/popper/gherkin/GherkinExtension.java [hunk 1]
```diff
@@ -53,18 +53,19 @@ public void beforeAll(ExtensionContext context) throws Exception {
 
     @Override
     public void beforeEach(ExtensionContext context) throws Exception {
-        getOrCreateRunner(context).startMethod(context.getRequiredTestInstance(), context.getRequiredTestMethod());
+        getOrCreateRunner(context).startMethod(context, context.getRequiredTestInstance(),
+                context.getRequiredTestMethod());
```
Comment (mibutec) [inline src/main/java/org/popper/gherkin/GherkinExtension.java]: Just providing the parameter context would be fine here

File: src/main/java/org/popper/gherkin/GherkinExtension.java [hunk 2]
```diff
@@ -53,18 +53,19 @@ public void beforeAll(ExtensionContext context) throws Exception {
...
     }
 
     @Override
     public void afterEach(ExtensionContext context) throws Exception {
-        getOrCreateRunner(context).endMethod(context.getRequiredTestInstance(), context.getRequiredTestMethod(),
-                context.getExecutionException());
+        getOrCreateRunner(context).endMethod(context, context.getRequiredTestInstance(),
+                context.getRequiredTestMethod(), context.getExecutionException());
```
Comment (mibutec) [inline src/main/java/org/popper/gherkin/GherkinExtension.java]: Just providing the parameter context would be fine here

**Review by mibutec — CHANGES_REQUESTED** [review]
2 calls to GherkinRunner could be simplified

File: src/main/java/org/popper/gherkin/GherkinExtension.java (hunk 2 above)
Comment (ArneAlbrechtW) [reply src/main/java/org/popper/gherkin/GherkinExtension.java]: Ok. Fixed. Hope, if've done right with github.

**Review by mibutec — APPROVED** [review]

**Pull request merged**